kernel/src/main/java/org/kframework/compile/checks/CheckKLabels.java

```diff
@@ -149,6 +150,32 @@ public void check() {
                 kem.registerCompilerWarning(ExceptionType.UNUSED_SYMBOL, errors, "Symbol '" + symbol + "' defined but not used. Add the 'unused' attribute if this is intentional.", klabelProds.get(symbol));
             }
         }
+        for (KLabel function : iterable(mainMod.functions())) {
+            boolean allConcrete = true;
+            boolean allSymbolic = true;
+            for (Rule rule : iterable(mainMod.rulesFor().get(function).getOrElse(() -> Collections.<Rule>Set()))) {
+                if ((rule.att().getOptional(Att.CONCRETE()).orElse("foo").equals("") &&
+                    rule.att().contains(Att.SYMBOLIC())) ||
+                    (rule.att().getOptional(Att.SYMBOLIC()).orElse("foo").equals("") &&
+                    rule.att().contains(Att.CONCRETE()))) {
+                    errors.add(KEMException.compilerError("Rule cannot be both concrete and symbolic in the same variable.", rule));
+                }
+                if (!rule.att().getOptional(Att.CONCRETE()).orElse("foo").equals("")) {
+                    allConcrete = false;
+                }
+                if (!rule.att().getOptional(Att.SYMBOLIC()).orElse("foo").equals("")) {
+                    allSymbolic = false;
+                }
```

Comment:
I really am not a fan of this way of selecting `"foo"` as the default to compare too. Would prefer something like:

```
Set<String> concreteVars = computeConcreteVars();
Set<String> symbolicVars = computeSymbolicVars();
if ((! (concreteVars.isEmpty() && symbolicVars.isEmpty())) && hasSimplification()) {
   error("Can't have concrete/symbolic without simplification");
}
if (concreteVars.intersects(symbolicVars)) {
   error("Can't have the same variable both concrete/symbolic: " concreteVars.intersection(symbolicVars));
}
```

The logic you have here is pretty hard to read.

Reply:
I will refactor out a function with a more human-readable name, but note that we're not doing anything quite this complicated, nor do I want to implement that now when it was not specifically requested as part of this issue. We can create another issue and prioritize it next week if we want to add more advanced checks. I mostly just implemented this check as part of implementing the other one.